crates/grafbase-sdk/src/host_io/event_queue.rs

```diff
@@ -253,8 +299,8 @@ impl ExecutedSubgraphRequest {
     ///
     /// This includes both successful responses and various types of failures
     /// (e.g., rate limiting, server errors).
-    pub fn executions(&self) -> impl Iterator<Item = RequestExecution> {
-        self.0.executions.clone().into_iter().map(RequestExecution::from)
+    pub fn executions(&self) -> impl Iterator<Item = RequestExecution<'_>> {
```

Comment:
why not just consume self? the HTTP headers API became a lot more complex because of this. The events here are meant to be consumed, you would pass owned variants in pure Rust. The events would be worth writing explicitly instead of letting wasmtime bindgen generate it IMHO. This would avoid the wrapping issue and we could just expose structs with public members instead.

Reply:
The issue is the damn headers. We want to pass them as a resource, and this complicates things a lot.